### ISSUE TITLE
`S` should delete the line and start insert, but it doesn't delete the indent.

### ISSUE THREAD

**Issue by YangChenyuan** [VSCodeVim/Vim#6443]
If there is an indent in the start of the line, `S` doesn't delete the indent, it only deletes the text after the indent and start insert.
Just like this
![Peek 2021-04-07 14-00](https://user-images.githubusercontent.com/55501188/113817934-a812be80-97a9-11eb-99a3-62f9650b7c12.gif)

**Comment by J-Fields** [comment]
Yeah we currently delete the whole line and then tell the language server to reindent the cursor. If there's no active language server, though (or it doesn't support this function, like C++ and python currently), this doesn't work. We should probably do as you say (delete all but the indent) and then tell the language server to reindent.

**Comment by YangChenyuan** [comment]
> Yeah we currently delete the whole line and then tell the language server to reindent the cursor. If there's no active language server, though (or it doesn't support this function, like C++ and python currently), this doesn't work. We should probably do as you say (delete all but the indent) and then tell the language server to reindent.

I got it. There are some characters of VSCode is confusing that it is tricky to handle every case.
